test/UnitTest/TestingRepositories/InstanceTests.cs

```diff
@@ -654,6 +694,106 @@ public async Task Instance_GetInstancesFromQuery_MatchFromPresentationFields_Ok(
             Assert.Equal(1, instances.Count);
         }
 
+        /// <summary>
+        /// Test GetInstancesFromQuery with CompleteConfirmations, primary org, match
+        /// </summary>
+        [Fact]
+        public async Task Instance_GetInstancesFromQuery_CompleteConfirmations_PrimaryOrg_Match_Ok()
+        {
+            // Arrange
+            Instance newInstance = TestData.Instance_1_1.Clone();
+            newInstance.CompleteConfirmations = new() { new() { StakeholderId = "TTD", ConfirmedOn = DateTime.Now } };
+            await _instanceFixture.InstanceRepo.Create(newInstance, CancellationToken.None);
+
+            InstanceQueryParameters queryParams = new()
+            {
+                Size = 100,
+                ExcludeConfirmedBy = new Microsoft.Extensions.Primitives.StringValues(["TTD"]),
+                Org = "TTD"
+            };
+
+            // Act
+            var instances = await _instanceFixture.InstanceRepo.GetInstancesFromQuery(queryParams, true, CancellationToken.None);
+
+            // Assert
+            Assert.Equal(0, instances.Count);
+        }
+
+        /// <summary>
+        /// Test GetInstancesFromQuery with CompleteConfirmations, primary org, no match
+        /// </summary>
+        [Fact]
+        public async Task Instance_GetInstancesFromQuery_CompleteConfirmations_PrimaryOrg_NoMatch_Ok()
+        {
+            // Arrange
+            Instance newInstance = TestData.Instance_1_1.Clone();
+            newInstance.CompleteConfirmations = new() { new() { StakeholderId = "TTD", ConfirmedOn = DateTime.Now } };
+            await _instanceFixture.InstanceRepo.Create(newInstance, CancellationToken.None);
+
+            InstanceQueryParameters queryParams = new()
+            {
+                Size = 100,
+                ExcludeConfirmedBy = new Microsoft.Extensions.Primitives.StringValues(["SKD"]),
+                Org = "TTD"
+            };
+
+            // Act
+            var instances = await _instanceFixture.InstanceRepo.GetInstancesFromQuery(queryParams, true, CancellationToken.None);
+
+            // Assert
+            Assert.Equal(1, instances.Count);
+        }
+
+        /// <summary>
+        /// Test GetInstancesFromQuery with CompleteConfirmations, other org, match
+        /// </summary>
+        [Fact]
+        public async Task Instance_GetInstancesFromQuery_CompleteConfirmations_OtherOrg_Match_Ok()
+        {
+            // Arrange
+            Instance newInstance = TestData.Instance_1_1.Clone();
+            newInstance.CompleteConfirmations = new() { new() { StakeholderId = "SKD", ConfirmedOn = DateTime.Now } };
+            await _instanceFixture.InstanceRepo.Create(newInstance, CancellationToken.None);
+
+            InstanceQueryParameters queryParams = new()
+            {
+                Size = 100,
+                ExcludeConfirmedBy = new Microsoft.Extensions.Primitives.StringValues(["SKD"]),
+                Org = "SKD"
```

Comment:
Should the org parameter be TTD?

The primary purpose of the Org parameter is to include instances for apps owned by the org. In this test the instance belong to TTD. The result should be 0 instances even without the ExcludeConfirmedBy parameter.

Reply:
Fixed